tools/pubsys/src/repo/check_expirations/mod.rs

```diff
@@ -0,0 +1,190 @@
+    end_date: DateTime<Utc>,
+) -> HashMap<tough::schema::RoleType, DateTime<Utc>>
+where
+    T: tough::Transport,
+{
+    let mut expirations = HashMap::new();
+    info!(
+        "Looking for metadata expirations happening from now to {}",
+        end_date
+    );
+    if repo.root().signed.expires <= end_date {
+        expirations.insert(tough::schema::RoleType::Root, repo.root().signed.expires);
+    }
+    if repo.snapshot().signed.expires <= end_date {
+        expirations.insert(
+            tough::schema::RoleType::Snapshot,
+            repo.snapshot().signed.expires,
+        );
+    }
+    if repo.targets().signed.expires <= end_date {
+        expirations.insert(
+            tough::schema::RoleType::Targets,
+            repo.targets().signed.expires,
+        );
+    }
+    if repo.timestamp().signed.expires <= end_date {
+        expirations.insert(
+            tough::schema::RoleType::Timestamp,
+            repo.timestamp().signed.expires,
+        );
+    }
+
+    expirations
+}
+
+fn check_expirations(
+    transport: &RepoTransport,
+    root_role_path: &PathBuf,
+    metadata_url: &Url,
+    targets_url: &Url,
+    expirations_between_now_and: Option<DateTime<Utc>>,
+) -> Result<()> {
+    // Create a temporary directory where the TUF client can store metadata
+    let workdir = tempdir().context(repo_error::TempDir)?;
+    let settings = Settings {
+        root: File::open(root_role_path).context(repo_error::File {
+            path: root_role_path,
+        })?,
+        datastore: workdir.path(),
+        metadata_base_url: metadata_url.as_str(),
+        targets_base_url: targets_url.as_str(),
+        limits: Limits::default(),
+        // We're gonna check the expiration ourselves
+        expiration_enforcement: ExpirationEnforcement::Unsafe,
+    };
+
+    // Load the repository
+    let repo = Repository::load(transport, settings).context(repo_error::RepoLoad {
+        metadata_base_url: metadata_url.clone(),
+    })?;
+    info!("Loaded TUF repo:\t{}", metadata_url);
+
+    info!("Root expiration:\t{}", repo.root().signed.expires);
+    info!("Snapshot expiration:\t{}", repo.snapshot().signed.expires);
+    info!("Targets expiration:\t{}", repo.targets().signed.expires);
+    info!("Timestamp expiration:\t{}", repo.timestamp().signed.expires);
+    // Check for upcoming metadata expirations if a timeframe is specified
+    if let Some(expiration_limit) = expirations_between_now_and {
+        let upcoming_expirations = find_upcoming_metadata_expiration(&repo, expiration_limit);
+        if !upcoming_expirations.is_empty() {
+            for (role, date) in upcoming_expirations {
+                if date < Utc::now() {
+                    error!("Repo '{}': '{}' expired on {}", metadata_url, role, date)
+                } else {
+                    warn!(
+                        "Repo '{}': '{}' expiring between now and {} on {}",
+                        metadata_url, role, expiration_limit, date
```

Comment:
```suggestion
                        "Repo '{}': '{}' expiring in {} at {}",
                        metadata_url, role, date - Utc::now(), date
```
(maybe a bit clearer; might want to save one Utc::now() before the loop)